### ISSUE TITLE
Error: not a valid address

### ISSUE THREAD

**Issue by cryptix** [ssbc/scuttlebot#299]
I started a new patchwork like this to demo it in our institute network:

`patchwork --allowRemoteAccess --unsafe --host svaHacking`

Where `svaHacking` is a `/etc/hosts` alias. It crashed my laptop instance with this:

```
/home/cryptix/patchwork/node_modules/scuttlebot/plugins/gossip.js:103
          throw new Error('not a valid address:' + JSON.stringify(addr))
          ^

Error: not a valid address:{"host":"svaHacking","port":8008,"key":"@U1+BrfWhx8x3RSCJuRDAaKCnPYsQdQRyGMKncWl1Y1o=.ed25519"}
    at Object.<anonymous> (/home/cryptix/patchwork/node_modules/scuttlebot/plugins/gossip.js:103:17)
    at apply (/home/cryptix/patchwork/node_modules/muxrpc-validation/index.js:173:15)
    at Object.<anonymous> (/home/cryptix/patchwork/node_modules/muxrpc-validation/index.js:62:14)
    at Object.hooked [as add] (/home/cryptix/patchwork/node_modules/hoox/index.js:10:15)
    at Socket.<anonymous> (/home/cryptix/patchwork/node_modules/scuttlebot/plugins/local.js:19:19)
    at emitOne (events.js:90:13)
    at Socket.emit (events.js:182:7)
    at Socket.<anonymous> (/home/cryptix/patchwork/node_modules/broadcast-stream/index.js:45:12)
    at emitTwo (events.js:100:13)
    at Socket.emit (events.js:185:7)
```

It's a bit fringe but I was quite surprised to see it crash like that.

**Comment by pfraze** [comment]
Humm we must have a dumb validator somewhere in there

**Comment by clehner** [comment]
util.[isAddress](https://github.com/ssbc/scuttlebot/blob/master/lib/util.js#L93) checks if the hostname is a (fully qualified) domain name. In what cases is this check needed? Should isDomain be replaced with a regex check that doesn't require dots in the domain name?

**Comment by pfraze** [comment]
I'm wondering if the hostname segment needs validation at all, except maybe to make sure it's a non-empty string?

**Comment by dominictarr** [comment]
this is fixed in 7.6.4
